video/out/opengl/context.c

```diff
@@ -270,6 +270,9 @@ bool ra_gl_ctx_start_frame(struct ra_swapchain *sw, struct ra_fbo *out_fbo)
     struct priv *p = sw->priv;
     out_fbo->tex = p->wrapped_fb;
     out_fbo->flip = !p->params.flipped; // OpenGL FBs are normally flipped
+
+    if (p->params.start_frame)
+        p->params.start_frame(sw->ctx);
```

Comment:
Would be nice if this could be avoided.

Reply:
I don't see how to avoid this, the atmoic request must be share between on the opengl rendering loop and the hwdec. The atomic request needs to be created before each frame and put in the native struct to be shared. 
Do you have any better place in mind where I could do this ?

Reply:
Why can't you do it in ra_swapchain_fns.start_frame?

Reply:
Yeah I agree, didn't have noticed the swapchain. By the way looking at how this works, I'm wondering if it wouldn't be cleaner in the `context_drm_egl.c` to have different `ra_swapchain_fns` (one for drm legacy and one for drm atomic), set them accordingly in the `init`function depending on what's supported, and just have the code splitted in different functions. what would be your thought on this ?

Reply:
Sounds ok too.